docs/deploy.md

```diff
@@ -0,0 +1,66 @@
+# Deploying Cluster API Provider Openshift BareMetal
+
+
+## Deploy on vanilla Kubernetes
+This provider is configured to deploy on [Red Hat OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift) (OCP) by default.
+
+To deploy this provider on vanilla Kubernetes, the following must be done:
+
+### Prerequisites
+
+A vanilla Kubernetes such as [Kind](https://kind.sigs.k8s.io/) must be deployed and you must be authenticated to the cluster.
+
+The following services are required on your cluster before installing this provider.
+
+1. Install [Assisted-Service operator](https://github.com/openshift/assisted-service/blob/master/docs/dev/operator-on-kind.md)
+2. Install [CAPI](https://cluster-api.sigs.k8s.io/user/quick-start.html)
+3. Install [CAPM3](https://book.metal3.io/capm3/installation_guide)
+
+The following CLI tools are required:
+
+1. [kustomize](https://kubectl.docs.kubernetes.io/installation/kustomize/)
+2. [kubectl](https://kubernetes.io/docs/tasks/tools/)
+
+### Deploy providers
+
+Clone this github repository
+
+```bash
+git clone http://github.com/openshift-assisted/cluster-api-agent
+```
+
+Append the following to the bottom of the manager [kustomization.yaml file](../config/manager/kustomization.yaml)
```

Comment:
Should we suggest the remote resource approach instead of checking out the whole repo?
i.e.

```
cat <<EOF > mytest/myfile
heredoc> apiVersion: kustomize.config.k8s.io/v1beta1
heredoc> kind: Kustomization
heredoc> metadata:
  name: internal-image-url
resources:
- https://raw.githubusercontent.com/openshift-assisted/cluster-api-agent/master/bootstrap/config/manager/manager.yaml
...
```

Reply:
Good idea, modified!